src/Documents.jl

```diff
@@ -307,9 +303,99 @@ function Document(;
+            plugin isa Plugin ||
+                throw("$(typeof(plugin)) is not a subtype of `Documenter.Plugin`.")
+            haskey(plugin_dict, typeof(plugin)) &&
+                throw("only one copy of $(typeof(plugin)) may be passed.")
+            plugin_dict[typeof(plugin)] = plugin
+        end
+    end
+
+    # html keywords depreciation
+    html_keywords = Dict()
+    if html_prettyurls !== nothing
+        Base.depwarn("""The `html_prettyurls` keyword argument has been moved to the
+            `HTML` plugin. To fix this warning, replace
+            ```
+                html_prettyurls = ...,
+            ```
+            with
+            ```
+                HTML(prettyurls = ...),
+            ```
+            """, :deploydocs)
+        html_keywords[:prettyurls] = html_prettyurls
+    end
+    if html_disable_git !== nothing
+        Base.depwarn("""The `html_disable_git` keyword argument has been moved to the
+            `HTML` plugin. To fix this warning, replace
+            ```
+                html_disable_git = ...,
+            ```
+            with
+            ```
+                HTML(disable_git = ...),
+            ```
+            """, :deploydocs)
+        html_keywords[:disable_git] = html_disable_git
+    end
+    if html_edit_branch !== nothing
+        Base.depwarn("""The `html_edit_branch` keyword argument has been moved to the
+            `HTML` plugin. To fix this warning, replace
+            ```
+                html_edit_branch = ...,
+            ```
+            with
+            ```
+                HTML(edit_branch = ...),
+            ```
+            """, :deploydocs)
+        html_keywords[:edit_branch] = html_edit_branch
+    end
+    if html_canonical !== nothing
+        Base.depwarn("""The `html_canonical` keyword argument has been moved to the
+            `HTML` plugin. To fix this warning, replace
+            ```
+                html_canonical = ...,
+            ```
+            with
+            ```
+                HTML(canonical = ...),
+            ```
+            """, :deploydocs)
+        html_keywords[:canonical] = html_canonical
+    end
+    if !isempty(html_keywords)
+        if Documenter.Writers.HTMLWriter.HTML in keys(plugin_dict)
+            throw("Documenter received both an `HTML` plugin and `html_*` keywords.")
+        end
+
+        plugin_dict[Documenter.Writers.HTMLWriter.HTML] =
+            Documenter.Writers.HTMLWriter.HTML(; html_keywords...)
+    end
+
+    Document(user, internal, plugin_dict)
+end
+
+"""
+    getplugin(doc::Document, T)
+
+Retrieves the [`Plugin`](@ref) type for `T` stored in `doc`. If `T` was passed to
+[`makedocs`](@ref), the passed type will be returned. Otherwise, a new `T` object
+will be created using the default constructor `T()`.
+"""
+function getplugin(doc::Document, plugin_type::Type{T}) where T <: Plugin
+    if !haskey(doc.plugins, plugin_type)
+        doc.plugins[plugin_type] = plugin_type()
```

Comment:
Wouldn't it be better to return `nothing` here instead of assuming that `T` has a `T()` constructor?

Reply:
The rational was to make the interface as easy as possible from the user's perspective. With this implementation, a user can just `using SomePlugin` in `make.jl` and things will Just Work (tm).